### ISSUE TITLE
Check if manga exists in the downloads folder and is yet unmanaged

### ISSUE THREAD

**Issue by Pokerface704** [keiyoushi/tachiyomi-extensions-source#6337]
### Source name

E-Hentai

### Source language

English

### Describe your suggested feature

Sometimes it is quality wise better to download a manga over torrent. This results (after adjusting to the Mihon structure) in a folder + Chapter.zip **WITHOUT** metadata. It would be great, if one could move the torrented manga to mihon/downloads/E-Hentai and your extension would automatically notice, that a manga has been added; thus synchronizing Mihons database and the locally (not yet managed) mangas.

I've come up with a rough feature I would have in mind for that:

After a search in e-hentai application, check the retrieved manga names in mihon/downloads/E-Hentai/:
(0. Check if manga is present in the E-Hentai folder AND is not managed. If both cases true, continue with 1. Otherwise abort)
1. Attach retrieved metadata to local manga
2. Mark manga as added to the library
3. Mark the chapter "Chapter(.zip)" as downloaded

(Optional, supporting) feature: Implement a button (or "on startup cycle"), that initiates a scan of mihon/downloads/E-Hentai/ folder. If a new, unmanaged manga is detected, look up the manga based on the folder name and apply the 4 previously mentioned steps.

Thank you for your work and the development of amazing extensions!

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] If this is an issue with the app itself, I should be opening an issue in the [app repository](https://github.com/mihonapp/mihon/issues/new/choose).
- [X] I have updated the app to version **[0.15.3](https://github.com/mihonapp/mihon/releases/latest)**.
- [X] I will fill out all of the requested information in this form.

### 

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/keiyoushi/extensions-source/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by Norsze** [comment]
This sounds like it is outside the scope of an extension

**Comment by Norsze** [comment]
You literally cause your own issue, so put in the extra time and download a chapter the normal way, extract the metadata file from the .zip and attach it to your download and put it in local. This would require app changes not just extension changes and no one will bother with such a niche case. Do the extra work or host it in a self host solution to get exactly what you want, much better if you are torrenting stuff.

Fopr you to read and do your own work:
https://mihon.app/docs/guides/local-source/

**Comment by Pokerface704** [comment]
Hey, thanks for your quick reply.
Based on your link to [https://mihon.app/docs/guides/local-source/](https://mihon.app/docs/guides/local-source/) I've assembled a rudimentary pipeline that reattaches the required metadata to the downloaded archives.

If anyone else encounters the, as you correctly mentioned, niche case (😄); I've published a repository with the Python code to generate Mihon's required local file structure from loose manga archives and manga folders. See [Pokerface704/Rudimentary-ExHentai-Torrent-and-ComicInfo.xml-Pipeline](https://github.com/Pokerface704/Rudimentary-ExHentai-Torrent-and-ComicInfo.xml-Pipeline/tree/main)
